### DISCUSSION TITLE
Two decimal places on air pressure when set to in Hg

### DISCUSSION THREAD

**Discussion by mi7wm** [breezy-weather/breezy-weather#1089] [Ideas]
### Describe your idea

I checked existing issues and couldn't find anything that seems like what I am requesting.

My preferred unit for air pressure is inHg (since that is what is used in aviation), and I was wondering if we could get two decimal places of specificity on the reading. Currently, the reading only goes to one decimal place, and that makes the unit harder to use, since the maximum useful range of this unit of measurement only goes from about 29.7 to 31.0.

Example:
Currently, air pressure appears like 29.9 inHg.
But it would be better if it stated 29.96 or 29.92 or 29.90 inHg.

I'd much appreciate it if this could be changed in this way. Thanks!

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.4 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Out of curiosity, what's your use case where 29.96 and 29.92 makes a difference? And what kind of difference these two values make? I was under the impression that for the general public, the most important thing was to see the variations and how far we are from the "normal" pressure.

~~Because if it's just to see more closely the variations, I guess we could put the min and max bounds of the line chart as 26 inHg to 32 inHg (which are world records) instead, or complementary.~~ Nevermind, it's already bounded to min and max values of the period

**Comment by papjul** [comment]
Alright, switching from hPa to inHg and I get what's the issue now. Fixed (and for other pressure units as well), and will be available in next version v5.2.5

**Reply by mi7wm** [reply]
Thanks! Just to clarify, I have attached a screenshot showing where in the app I'm looking. To answer your above question, it's just the degree of specificity needed to make use of the numbers. 29.92 inHg is normal atmospheric pressure. Something like 30.01 would be a high pressure reading, and 29.78 would be a very low pressure reading.
![image](https://github.com/breezy-weather/breezy-weather/assets/29791502/1ff0dda9-d987-469f-ad55-5fda3ff4271a)

**Reply by papjul** [reply]
> Just to clarify, I have attached a screenshot showing where in the app I'm looking

The fix is not limited to this area, but app-wide.

> 29.92 inHg is normal atmospheric pressure. Something like 30.01 would be a high pressure reading, and 29.78 would be a very low pressure reading.

Your examples show that 1 decimal is enough.

**Reply by mi7wm** [reply]
> Your examples show that 1 decimal is enough.

Does the current one decimal setup follow normal rounding rules? 29.94 will round to 29.9, 29.96 will round to 30?

Respectfully, if 29.95 rounds to 30, then I have a reading on the app that would indicate high pressure when it's actually closer to normal pressure at 29.92.

The metric is still useful in most circumstances with one decimal place of precision, but two would be optimal.

> The fix is not limited to this area, but app-wide.

If I may, what is the fix that has been applied, if not to change the metric to two decimal places?

Thanks for all your work on this project,
M

**Reply by papjul** [reply]
The change is visible here: 7ddb84a9b734f6c44696fae3d9946aedf1e0ad9b

> if not to change the metric to two decimal places

I didn't say that. I said that the change would be everywhere, not just in the place you showed (current pressure).
